platforms/posix/src/main.cpp

```diff
@@ -578,26 +578,30 @@ void print_usage()
 bool is_already_running(int instance)
 {
 	const std::string file_lock_path = std::string(LOCK_FILE_PATH) + '-' + std::to_string(instance);
-	struct flock fl;
 	int fd = open(file_lock_path.c_str(), O_RDWR | O_CREAT, 0666);
 
 	if (fd < 0) {
+		close(fd);
 		return false;
 	}
 
+	struct flock fl {};
+
 	fl.l_type   = F_WRLCK;
+
 	fl.l_whence = SEEK_SET;
+
 	fl.l_start  = 0;
+
 	fl.l_len    = 0;
+
 	fl.l_pid    = getpid();
 
 	if (fcntl(fd, F_SETLK, &fl) == -1) {
 		// We failed to create a file lock, must be already locked.
 		if (errno == EACCES || errno == EAGAIN) {
 			return true;
 		}
-
-		return false;
```

Comment:
Removing the return here changes the result: `errno` is set to 0.

Reply:
You're right. I misunderstood the usage of errno at line 163.  I'll revert this change.